R/distribution_formulas.R

```diff
@@ -0,0 +1,356 @@
+      "{.code {family}} is the incorrect type",
+      "*" = "family needs to be specified as a family class",
+      "*" = " e.g. `family = gaussian()`."))
+  }
+  if (!(family[["family"]] %in% families)) {
+    fam_name <- family[["family"]]
+    cli::cli_abort(c(
+      "FIMS currently does not offer the family, {.code {fam_name}}. for this distribution type",
+      "*" = "The families available for this distribution type are:",
+      "*" = "{.code {families}}."))
+
+  }
+  if(exists("data_type")){
+
+    if ((data_type == "agecomp" || data_type == "lengthcomp") &&
+        (family[["family"]] == "lognormal" || family[["family"]] == "gaussian")) {
+      fam_name <- family[["family"]]
+      cli::cli_abort(c(
+        "The family, {.code {fam_name}} is not available for the data type, {.code {data_type}}",
+        "*" = "The available families for this data type are:",
+        "*" = "`multinomial`."
+      ))
+    }
+    if ((data_type == "index" || data_type == "cpue") &&
+        family[["family"]] == "multinomial") {
+      fam_name <- family[["family"]]
+      cli::cli_abort(c(
+        "The family, {.code {fam_name}} is not available for the data type, {.code {data_type}}",
+        "*" = "The available families for this data type are:",
+        "*" = "`lognormal`, `gaussian`."
+      ))
+    }
+
+  }
+
+  if(!all(sd$value > 0)) {
+    sd_value <- sd$value
+    cli::cli_abort(c(
+    "At least one value for sd, {.code {sd_value}} is out of bounds.",
+    "*" = "All standard deviation values need to be positive."))
+  }
+
+  if(length(sd$estimated) > 1){
+    if(length(sd$value) != length(sd$estimated)){
+      sd_length <- length(sd$value)
+      est_length <- length(sd$estimated)
+      cli::cli_abort(c(
+        "The length of `sd$value` is {.code {sd_length}} and the length of `sd$estimated` is {.code {est_length}}",
+        "*" = "the size of `value` and `estimated` must match."))
+    }
+  }
+
+}
+
+#' Return name of expected value
+#' 
+#' @param family A description of the error distribution and link function to
+#'   be used in the model. The argument takes a family class, e.g. [stats::gaussian(link = "identity")]. 
+#' @param data_type A single string specifying the type of data that the distribution
+#'   will be fit to. Options are listed in the function call, where the first
+#'   option listed, i.e., `"index"` is the default.
+#' @return A string specifying the name of the expected value.
+#' 
+set_expected_name <- function(family, data_type){
+  expected_name <- NA
+  if(data_type == "index" || data_type == "cpue"){
+    if(family[["family"]] == "lognormal" || family[["family"]] == "gaussian"){
+      if(family[["link"]] == "log"){
+        expected_name <- "log_expected_index"
+      } 
+      if(family[["link"]] == "identity"){
+        expected_name <- "expected_index"
+      } 
+    }
+  }
+  if(data_type == "agecomp" || data_type == "lengthcomp"){
+    expected_name <- "proportion_catch_numbers_at_age"
+  }
```

Comment:
Only if statements are present, no else statement. So what happens if you do not fit inside an if statement?

Reply:
data_type uses match_args to set the value from the subset: `index, cpue, agecomp, lengthcomp`. A different data type will throw an error before this part of the code is run. Also, the default expected_name is NA, so technically that is the returned value if input does not fit inside the if statements.

Reply:
I just read that match_arg uses partial matching, so I added an additional validity check to make sure data_type is one of the four options available. I also realized we don't have checks on link functions. sdmTMB adds these checks to the family functions themselves. I added a check to throw an error if the expected_value is still NA while we work out where to put checks on link functions.